frb_rust/src/ffi/web.rs

```diff
@@ -0,0 +1,388 @@
+use super::DartAbi;
+use super::IntoDart;
+use super::MessagePort;
+pub use js_sys;
+pub use js_sys::Array as JsArray;
+use js_sys::*;
+use std::iter::FromIterator;
+pub use wasm_bindgen;
+pub use wasm_bindgen::closure::Closure;
+pub use wasm_bindgen::prelude::*;
+pub use wasm_bindgen::JsCast;
+use web_sys::BroadcastChannel;
+use web_sys::{DedicatedWorkerGlobalScope, Worker};
+
+pub trait IntoDartExceptPrimitive: IntoDart {}
+impl IntoDartExceptPrimitive for JsValue {}
+impl IntoDartExceptPrimitive for String {}
+impl<T: IntoDart> IntoDartExceptPrimitive for Option<T> {}
+
+mod pointer {
+    use std::ptr::NonNull;
+    use wasm_bindgen::convert::*;
+    use wasm_bindgen::describe::*;
+    use wasm_bindgen::JsValue;
+
+    /// A non-nullable pointer wrapping a [`NonNull`].
+    /// Represented in JS as an unsigned integer.
+    ///
+    /// Designed as a replacement for [`Option<*mut T>`].
```

Comment:
⚠️

Wait a bit... `NonNull<T>` for `Option<*mut T>`??? Then it means we will store `None` i.e. 0 inside `NonNull`? That would be a great problem :( It is `Option<NonNull<T>>` that seems similar to `*mut T`.

https://doc.rust-lang.org/std/ptr/struct.NonNull.html

> If you’re not sure if you should use NonNull<T>, just use *mut T!

Reply:
I'll need to rephrase this: it is a replacement for `*mut T` since `Option<*mut T>` is not allowed by wasm_bindgen right now, and will be removed once it becomes usable. 
(The first draft also had a `PhantomData<*mut T>` to make it invariant over T, but I must have dropped it for some reason.)

Reply:
> `Option<*mut T>` is not allowed by wasm_bindgen right now

Sure! Because `*mut T` is *allowed to be zero*, so it is not reasonable to write down `Option<*mut T>`. Instead, I guess we should always be careful `*mut T` may have zero (null pointer) in it. We can even use [is_null](https://doc.rust-lang.org/std/primitive.pointer.html#method.is_null) to check.

Btw, the https://doc.rust-lang.org/std/ptr/struct.NonNull.html has a bit explanations about `*mut T`.